src/main.cpp

```diff
@@ -102,8 +102,7 @@ main(int argc, char *argv[]) {
   task_pool_util::TaskPool::task_id_t force_shutdown = nullptr;
 
 #ifdef _WIN32
-  // Switch default C standard library locale to UTF-8 on Windows 10 1803+
-  setlocale(LC_ALL, ".UTF-8");
+  setlocale(LC_ALL, "C");
```

Comment:
Just linking the original PR for this code. Have we verified we no longer hit the issues linked to https://github.com/LizardByte/Sunshine/pull/1517 ?

Reply:
Current PR branch with Chinese characters printing to log is OK:
```diff
diff --git a/src/main.cpp b/src/main.cpp
index 7aaafbdd..07b6beda 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -200,6 +200,7 @@ main(int argc, char *argv[]) {
   // if anything is logged prior to this point, it will appear in stdout, but not in the log viewer in the UI
   // the version should be printed to the log before anything else
   BOOST_LOG(info) << PROJECT_NAME << " version: " << PROJECT_VER;
+  BOOST_LOG(info) << L"你好";

   if (!config::sunshine.cmd.name.empty()) {
     auto fn = cmd_to_func.find(config::sunshine.cmd.name);

```

PR with Chinese characters and Cameron's relevant change reverted crashes:
```diff
diff --git a/src/main.cpp b/src/main.cpp
index 7aaafbdd..bb6dbf68 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -108,7 +108,7 @@ main(int argc, char *argv[]) {
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
   // Use UTF-8 conversion for the default C++ locale (used by boost::log)
-  std::locale::global(std::locale(std::locale(), new std::codecvt_utf8<wchar_t>));
+  //std::locale::global(std::locale(std::locale(), new std::codecvt_utf8<wchar_t>));
 #pragma GCC diagnostic pop

   mail::man = std::make_shared<safe::mail_raw_t>();
@@ -200,6 +200,7 @@ main(int argc, char *argv[]) {
   // if anything is logged prior to this point, it will appear in stdout, but not in the log viewer in the UI
   // the version should be printed to the log before anything else
   BOOST_LOG(info) << PROJECT_NAME << " version: " << PROJECT_VER;
+  BOOST_LOG(info) << L"你好";

   if (!config::sunshine.cmd.name.empty()) {
     auto fn = cmd_to_func.find(config::sunshine.cmd.name);

```

Conclusion:
* The PR as-is will not break issues fixed by #1517
* #1517 never needed `setlocale` to be set to UTF-8, as it [wasn't](https://github.com/msys2/MINGW-packages/pull/20477#issuecomment-2027821426) [applying ](https://github.com/LizardByte/Sunshine/pull/2323#issuecomment-2027862071)on our MING64 builds.